src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -226,26 +226,48 @@ public async Task<bool> GetHasReservedNamespaceAsync(PackageURL purl, bool useCa
             return metadata;
         }
         
-        /// <inheritdoc />
-        public override async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool useCache = true)
+        /// <inheritdoc/>
+        public override async Task<bool> PackageExistsAsync(PackageURL purl, bool useCache = true)
         {
             Logger.Trace("PackageExists {0}", purl?.ToString());
             if (purl is null)
+            {
+                Logger.Trace("Provided PackageURL was null.");
+                throw new ArgumentNullException(nameof(purl), "Provided PackageURL was null.");
+            }
+
+            HttpClient httpClient = CreateHttpClient();
+            
+            // NuGet packages are case insensitive
+            string endpoint = $"{RegistrationEndpoint}{purl.Name.ToLowerInvariant()}/index.json";
```

Comment:
if they are case insensitive, is there a reason why you lowercase the name anyways? Seems like we shouldn't have to.

Reply:
The endpoints are all normalized to lowercase to reflect the fact that package names are case insensitive. E.g. https://api.nuget.org/v3-flatcontainer/**oWin**/index.json will return does not exist, when actually it should point to the same place as https://api.nuget.org/v3-flatcontainer/owin/index.json if NuGet package names are true to case insensitivity. So, we lowercase.